### PR TITLE
The logo can be changed if preferred

### PR DESCRIPTION
**Pull request by Koshila-cd**
A logo parameter is added

### CODE REVIEW

**Comment by dinukadesilva** [comment]
@Koshila-cd It's great work and glad about your intrest in contributing to app-header.

Please upload a screenshot of the new addition.

Also, it would be nice if you could rebase and update the author of all the commits. So, that all commits will be linked to your profile.

CC: @wyseguyonline @mhomolak

File: src/js/template.js [hunk 1]
```diff
@@ -17,17 +17,18 @@ function template (data, handlers, translate) {
   elementOpen("div", null, ["class", "o-header__container"])
     elementOpen("section", null, ["class", "o-header__section"])
       elementOpen("div", null, ["class", "o-header__brand"])
+      var logo = data.logo ? data.logo : images.logoLarge;
         if (data.mode === 'Basic' || data.mode === 'Legacy Course') {
           elementOpen("a", null, null, "href", data.links.home)
             elementOpen("div", null, ["class", "o-header__logo o-header__logo--pearson"])
-              elementOpen("img", null, null, "src", images.logoLarge, "alt", translate('Pearson - Home'))
+            elementOpen("img", null, null, "src", logo, "alt", translate('Pearson - Home'))
               elementClose("img")
```
Comment (dinukadesilva) [inline src/js/template.js]: I think you should not change this file manually and it has to be generated from `src/html/header.template`. Or have you missed to add `src/html/header.template` file to commit?
image

File: README.md [hunk 2]
```diff
@@ -136,13 +136,18 @@ Creates a new instance of `AppHeader`.
 | consoleBaseUrl           | `string`               | The consuming application base URL (default: https://console.pearson.com) |
 | links                    | `object`               | Links in the user dropdown. Refer to the [Links](#links) section for the default list  |
 | mode                     | `string`               | The mode. Refer to the [Modes](#modes) section for a list of supported modes and options. |
-
+| logo                     | `string`               | Logo in the left side of the header (default: Pearson logo) |
```
Comment (dinukadesilva) [inline README.md]: If you can add an example too. That's great.

File: src/js/template.js (hunk 1 above)
Comment (Koshila-cd) [reply src/js/template.js]: @dinukadesilva Thank you very much for the assistance. No it's not that I didn't commit the `header.template` file. I will change it accordingly and generate the code.

File: README.md (hunk 2 above)
Comment (Koshila-cd) [reply README.md]: @dinukadesilva I actually have added a description

image

And comment if it would not be enough.

**Comment by Koshila-cd** [comment]
Thanks for the assistance @dinukadesilva , and attached here is a screenshot of the logo change of the app-header.

image

@antelopeb @mhomolak

**Comment by Koshila-cd** [comment]
@dinukadesilva @antelopeb 
Can you verify what is wrong in my files for the test cases to fail? Or is it because i'm in a fork branch.

Thanks in Advance.

**Comment by antelopeb** [comment]
Yeah, it's because the environment variables aren't present, so it can't build properly. Will take a look now!

**Comment by antelopeb** [comment]
@Koshila-cd -- I believe the design requirement was to allow for a text string after the Pearson logo. Please see the designs from the UX team:
image

**Comment by dinukadesilva** [comment]
Also somethig to note is @antelopeb, There are some other teams having following style as well.
image


Either directing all the teams come to a one design could be an option, Or, what if both icons are considered as two customization icons. Company Logo and Application Logo.

**Comment by Koshila-cd** [comment]
@antelopeb  @dinukadesilva 

Here's the new change done to the app-header, the company logo and the application logo can be passed seperately.
image


Here I have given an example by adding the company logo.
image


Here I have added a screenshot of the requirement of my team "Statcrunch"
image

I have added an image input to the application logo since our team the application logo as it is.

Please clarify or suggest if any change is required.

**Comment by antelopeb** [comment]
@Koshila-cd -- does it still allow for a default behavior, where a team passes in nothing and it appears as it did before?

Sorry for the delay in reviewing. I will be trying to do this in the next few days.

**Comment by Koshila-cd** [comment]
@antelopeb 
No worries!

Yes the default logo is shown when the user doesn't pass any logo and let me show some screenshots: 

**1) When user doesn't pass any logo**

image


**2) When user passes one logo (Most probably the short Pearson logo)**

image

**3) When user passes both logos (Company logo and application logo)**

image


The above logos are images.

**There are some more changes to be done, you can start reviewing when I have completed doing changes. (By end of today (Sri Lankan Time)).

**Comment by antelopeb** [comment]
@Koshila-cd Can you also check for full responsiveness in your code? Sorry to keep throwing things at you, but this is looking great so far!! If you can do the responsive piece I should be able to merge quickly!!

Reference to design: https://uxframework.pearson.com/c/special-application-header/

**Comment by Koshila-cd** [comment]
@antelopeb The Mobile responsiveness for the Pearson logo is added, i.e when the user adds an application logo, the Pearson text of the default Pearson logo is removed from Width 768px and below, and only the Small Pearson(P) icon will be remaining.

Width = 768px :
image

Width = 767px : 
image

**Comment by davodey** [comment]
@Koshila-cd can you look at the Travis build and see if you can fix the errors?  Also, I am having issues trying to run the dev and build scripts, I get errors.  I'm not sure how long it's been like this, so please let me know if I need to disregard, but it seems like something we should have fixed.

CC @antelopeb

**Comment by Koshila-cd** [comment]
@davodey The errors of Travis build is probably because I don't have access to the repository or I'll have a look in it. And about the dev and build errors, if you can post a screenshot then maybe I can have a look in it.
Thanks.

CC: @antelopeb

**Comment by davodey** [comment]
@Koshila-cd it looks like the script errors are in master.  Im reaching out to others to see if its something we need to be concerned with.

**Pull request merged**